src/pages/students/edit-student/EditStudent.tsx

```diff
@@ -0,0 +1,135 @@
+import StudentData, { type StudentDataProps } from "../forms/studentDataForm/StudentData"
+import PaymentData, { type PaymentDataProps } from "../forms/paymentDataForm/PaymentData"
+import { useGetStudentByIdQuery, useUpdateStudentMutation } from "../../../app/services/StudentService"
+import useAuthentication from "../../../hooks/useAuthentication";
+import { useRef, type JSX } from "react";
+//import PlanData, { type PlanDataProps } from "../forms/planDataForm/PlanData";
+import { useLocation, useNavigate } from "react-router";
+import { useParams } from "react-router-dom";
+import { MainContainer, TitleContainer, Title, ButtonsContainer, Button, FormContainer } from "./EditStudent.styles";
+import { formatDateToDash, formatDateToISO } from "../../../utils/DateFormatter";
+import toast from "react-hot-toast";
+import type { UpdateStudentRequest } from "../../../app/types/requests/UpdateStudentRequest.type";
+import BackIcon from "../../../assets/back-icon.svg";
+
+
+function EditStudent() {
+
+    const { userId } = useAuthentication();
+
+    const formStudentDataRef = useRef<any>(undefined);
+
+    const formPaymentDataRef = useRef<any>(undefined);
+
+    const { numberOfStep } = useParams();
+
+    const { studentId } = useLocation().state;
+
+    const numberOfStepNum = numberOfStep ? parseInt(numberOfStep) : null;
+
+    const { data: studentSaveData } = useGetStudentByIdQuery(studentId!);
+
+    const [updateStudent] = useUpdateStudentMutation();
+
+    const navigate = useNavigate();
+
+    const showStudentData = (studentSaveData: StudentDataProps) => {
+
+        const formattedBirthday = formatDateToISO(studentSaveData.birthday);
+
+        return {
+            name: studentSaveData.name,
+            lastName: studentSaveData.lastName,
+            dni: studentSaveData.dni,
+            cellphoneNumber: studentSaveData.cellphoneNumber,
+            birthday: formattedBirthday,
+            pathologies: studentSaveData.pathologies ?? undefined,
+        };
+    }
+
+    if (!studentSaveData) {
+        return <p>Cargando datos del estudiante...</p>;
+    }
+
+    const handleStudentData = (data: StudentDataProps) => {
```

Comment:
Los metodos quedarian mejor si se llamaran algo como esto

```suggestion
    const handleUpdateStudentData = (data: StudentDataProps) => {
```